### PR TITLE
Update SDK to latest OpenAPI spec

### PR DESCRIPTION
**Pull request by tanmay-db**
## Changes

- Update SDK to OpenAPI spec
- Also, fix naming convention, we use Pascal case (.PascalName) instead of Camel case (.Name). Because of this we now generate the following:
`CSPEnablement` -> `CspEnablement`

## Tests

Unit tests, integration test will run release PR. 
- [ ] `make test` run locally
- [ ] `make fmt` applied
- [ ] relevant integration tests applied

### CODE REVIEW

File: .codegen/service.py.tmpl [hunk 1]
```diff
@@ -101,7 +101,7 @@ class {{.PascalName}}{{if eq "List" .PascalName}}Request{{end}}:{{if .Descriptio
 {{- end -}}
 
 {{range .Services}}
-class {{.Name}}API:{{if .Description}}
+class {{.PascalName}}API:{{if .Description}}
```
Comment (tanmay-db) [inline .codegen/service.py.tmpl:104]: Highlight

File: docs/gen-client-docs.py [hunk 2]
```diff
@@ -388,7 +388,7 @@ def _get_tag_name(self, class_name, service_name) -> Tag:
             class_name = class_name.replace('Ext', 'API')
         class_name = class_name[:-3]
         for tag_name, t in self.mapping.items():
-            if t.service == class_name:
+            if t.service.lower() == str(class_name).lower():
```
Comment (tanmay-db) [inline docs/gen-client-docs.py:391]: Highlight

Comment (mgyucht) [reply docs/gen-client-docs.py:391]: Note that the SDK has tools to manipulate string casing: https://github.com/databricks/databricks-sdk-py/blob/main/databricks/sdk/casing.py. However, I think this is still safe enough as we shouldn't have services that are the same modulo case.

File: .codegen/service.py.tmpl (hunk 1 above)
Comment (mgyucht) [reply .codegen/service.py.tmpl:104]: OK, as this is aligned with the treatment of this in Go SDK.

**Review by mgyucht — APPROVED** [review]
LGTM. We need to announce in this release the change to the settings APIs.

**Pull request merged**